src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,54 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $day-gap;
+  padding: $calendar-padding;
+  width: $calendar-width;
+
+  @for $index from 28 through 30 {
+    &--length_#{$index} {
+      @for $index-2 from 1 through 3 {
+        & .calendar__day:nth-child(#{$index + $index-2}) {
+          display: none;
+        }
+      }
+    }
+  }
+
+  &__day {
+    @include square($square-size);
+    border: 1px black solid;
+    background-color: #eee;
+    font-family: Arial, "Helvetica Neue", Helvetica, sans-serif;
+    font-size: 30px;
+    text-align: center;
+    line-height: $square-size;
+    transition: all 0.5s ease;
+
```

Comment:
No need for transition all, use for transform and color only.
`transition: background-color 0.5s, transform 0.5s ease`